api/src/main/java/uk/gov/dvsa/mot/app/logging/JsonLayout.java

```diff
@@ -0,0 +1,169 @@
+package uk.gov.dvsa.mot.app.logging;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.ThreadContext.ContextStack;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.AbstractStringLayout;
+import org.apache.logging.log4j.message.MapMessage;
+import org.apache.logging.log4j.message.Message;
+import org.codehaus.jackson.JsonFactory;
+import org.codehaus.jackson.JsonGenerator;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+@Plugin(name = "JSONLayout", category = "Core", elementType = "layout", printObject = true)
+public class JsonLayout extends AbstractStringLayout {
+    public static final String AwsRequestIdKey = "AWSRequestId";
```

Comment:
do we need to couple JSONLayout with the specific attribute names ? It should remain agnostic as to what content it handles.